rskj-core/src/test/java/co/rsk/peg/union/UnionStorageProviderImplTest.java

```diff
@@ -0,0 +1,212 @@
+        // Act
+        Optional<RskAddress> unionBridgeAddress = unionBridgeStorageProvider.getAddress();
+
+        // Assert
+        assertTrue(unionBridgeAddress.isEmpty());
+    }
+
+    @Test
+    void getAddress_whenAddressStored_shouldReturnStoredAddress() {
+        // Arrange
+        RskAddress rskAddress = TestUtils.generateAddress("address");
+
+        // Arrange
+        storageAccessor.saveToRepository(
+            UnionBridgeStorageIndexKey.UNION_BRIDGE_CONTRACT_ADDRESS.getKey(),
+            rskAddress,
+            BridgeSerializationUtils::serializeRskAddress
+        );
+
+        // Act
+        Optional<RskAddress> unionBridgeAddress = unionBridgeStorageProvider.getAddress();
+
+        // Assert
+        assertTrue(unionBridgeAddress.isPresent());
+        assertEquals(rskAddress, unionBridgeAddress.get());
+    }
+
+    @Test
+    void getAddress_whenAddressSet_shouldReturnCacheAddress() {
+        // Arrange
+        RskAddress rskAddress = TestUtils.generateAddress("address");
+        unionBridgeStorageProvider.setAddress(rskAddress);
+
+        // Act
+        Optional<RskAddress> unionBridgeAddress = unionBridgeStorageProvider.getAddress();
+
+        // Assert
+        assertTrue(unionBridgeAddress.isPresent());
+        assertEquals(rskAddress, unionBridgeAddress.get());
+    }
+
+    @Test
+    void getAddress_whenAddressSetAndStored_shouldReturnSetAddress() {
+        // Arrange
+        RskAddress rskAddress = TestUtils.generateAddress("address");
+        unionBridgeStorageProvider.setAddress(rskAddress);
+        unionBridgeStorageProvider.save();
+
+        // Act
+        Optional<RskAddress> unionBridgeAddress = unionBridgeStorageProvider.getAddress();
+
+        // Assert
+        assertTrue(unionBridgeAddress.isPresent());
+        assertEquals(rskAddress, unionBridgeAddress.get());
+    }
+
+    @Test
+    void setAddress_whenEmptyAddress_shouldNotStore() {
+        // Arrange
+        RskAddress rskAddress = new RskAddress(new byte[20]);
+
+        // Act
+        unionBridgeStorageProvider.setAddress(rskAddress);
+        unionBridgeStorageProvider.save();
+
+        // Assert
+        RskAddress retrievedAddress = storageAccessor.getFromRepository(
+            UnionBridgeStorageIndexKey.UNION_BRIDGE_CONTRACT_ADDRESS.getKey(),
+            BridgeSerializationUtils::deserializeRskAddress
+        );
+        assertNull(retrievedAddress);
+    }
+
+    @Test
+    void setAddress_whenNull_shouldNotStore() {
+        // Act
+        unionBridgeStorageProvider.setAddress(null);
+        unionBridgeStorageProvider.save();
+
+        // Assert
+        RskAddress retrievedAddress = storageAccessor.getFromRepository(
+            UnionBridgeStorageIndexKey.UNION_BRIDGE_CONTRACT_ADDRESS.getKey(),
+            BridgeSerializationUtils::deserializeRskAddress
+        );
+        assertNull(retrievedAddress);
+    }
+
+    @Test
+    void setAddress_whenAddressSet_shouldNotStore() {
```

Comment:
```suggestion
    void setAddress_withoutSaving_shouldNotStore() {
```

something like that?

Reply:
Updated